src/components/ColorPicker/components/TextPicker/tests/TextPicker.test.tsx

```diff
@@ -0,0 +1,153 @@
+import React from 'react';
+import {mountWithAppProvider} from 'test-utilities/legacy';
+import {
+  expandHex,
+  rgbStringToHex,
+  nameToHex,
+  normalizeColorString,
+} from '../../../../../utilities/color-transformers';
+import {TextPicker} from '../TextPicker';
+import {TextField} from '../../../../TextField';
+
+const red = {
+  hue: 0,
+  saturation: 1,
+  brightness: 1,
+};
+
+describe('<TextPicker />', () => {
+  const mockProps = {
+    color: red,
+    allowAlpha: false,
+    onChange: noop,
+  };
+
+  describe('color', () => {
+    it("displays #FF0000 when 'red' color is passed as prop", () => {
+      const textPicker = mountWithAppProvider(<TextPicker {...mockProps} />);
+      const textField = textPicker.find(TextField);
+      const expectedHex = '#FF0000';
+
+      expect(textField.prop('value')).toBe(expectedHex);
+    });
+  });
+
+  describe('allowAlpha', () => {
+    it('renders color swatch', () => {
+      const textPicker = mountWithAppProvider(<TextPicker {...mockProps} />);
+
+      expect(textPicker.find('div.TextFieldSwatch')).toHaveLength(1);
+    });
+  });
+
+  describe('onChange()', () => {
+    it("calls onChange with #FFFFFF when 'white' is entered", () => {
+      const onChangeSpy = jest.fn();
+      const textPicker = mountWithAppProvider(
+        <TextPicker {...mockProps} onChange={onChangeSpy} />,
+      );
+      const textField = textPicker.find(TextField);
+      const colorValue = 'white';
+      const expectedHex = nameToHex(normalizeColorString(colorValue));
+
+      (textField.find('input') as any).instance().value = colorValue;
```

Comment:
Same comment about getDOMNode and questions about blur

Reply:
Answer about `blur` above, will change to `getDOMNode`.

Reply:
`getDOMNode` fails without casting, so I will change (this and all other instances) to 
```
(textField.find('input').getDOMNode() as HTMLInputElement).value = colorValue;
```

![https://screenshot.click/07-17-hsngh-0nz7a.png](https://screenshot.click/07-17-hsngh-0nz7a.png)

Reply:
For reference a way to avoid casting here could be to use a type guard. Here's an example using [instanceof](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/instanceof) 😄 
```tsx
const input = textField.find(‘input’).getDOMNode();
const value = input instanceof HTMLElement ? ‘’ : input.value;
```

Reply:
🤔 Using the ternary operator here gives the further error of 
`Tests should not contain conditional statements.eslint(jest/no-if)`

It also didn't seem to allow me to set the input node's value.

Reply:
@AndrewMusgrave I pushed a fix using casting (`(textField.find('input').getDOMNode() as HTMLInputElement).value`) just to get this to a place where it is ready for another review, I'm not ignoring your advice, don't worry! Just haven't been able to get an error free method to work without casting 🤔